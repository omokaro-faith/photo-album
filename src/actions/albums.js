import axios from 'axios';
import swal from 'sweetalert';
import { ERROR_OCCURED } from '../constants/constants';

export const setAlbums = albums => ({
  type: 'GET_ALBUMS',
  albums,
});

export const setAlbumsError = () => ({
  type: 'GET_ALBUMS_ERROR',
  message: 'Cannot fetch albums'
});

export const setAlbumsLength = totalAlbums => ({
  type: 'GET_ALL_ALBUMS',
  totalAlbums,
});

export const fetchAllAlbumsError = () => ({
  type: 'GET_ALL_ALBUMS_ERROR',
  errorMessage: 'Cannot fetch all albums'
});

export const fetchAllAlbums = () => async (dispatch) => {
  try {
    const albums = await axios.get('https://jsonplaceholder.typicode.com/albums');
    return dispatch(setAlbumsLength(albums.data.length));
  } catch (err) {
    const { error, status, buttons } = ERROR_OCCURED;
    swal(error, 'Cannot fetch all albums', status, {
				buttons,
		});
    return(dispatch(fetchAllAlbumsError()));
  }
}

export const getAlbums = (start = 0, limit) => async (dispatch) => {
  try {
    const albums = await axios.get(`https://jsonplaceholder.typicode.com/albums?_start=${start}&_limit=${limit}`);
    return dispatch(setAlbums(albums.data));
  } catch (err) {
    const { error, status, buttons } = ERROR_OCCURED;
    swal(error, 'Cannot fetch albums', status, {
				buttons,
		});
    return(dispatch(setAlbumsError()));
  }
};
