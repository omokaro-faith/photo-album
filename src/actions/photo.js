import axios from 'axios';

export const setPhotos = photos => ({
  type: 'GET_PHOTOS',
  photos,
});

export const setPhotoError = () => ({
  type: 'GET_PHOTOS_ERROR',
  message: true
});

export const setPhotosLength = totalPhotos => ({
  type: 'GET_ALL_PHOTOS',
  totalPhotos,
});

export const fetchAllPhotosError = () => ({
  type: 'GET_ALL_PHOTOS_ERROR',
  errorMessage: true
});

export const fetchAllPhotos = albumId => async (dispatch) => {
  try {
    const photos = await axios.get(`https://jsonplaceholder.typicode.com/photos?albumId=${albumId}`);
    return dispatch(setPhotosLength(photos.data.length));
  } catch (err) {
    return(dispatch(fetchAllPhotosError()));
  }
}

export const getPhotos = (albumId, start= 0 , limit) => async (dispatch) => {
  try {
    const photos = await axios.get(`https://jsonplaceholder.typicode.com/photos?albumId=${albumId}&_start=${start}&_limit=${limit}`);
    return dispatch(setPhotos(photos.data));
  } catch (err) {
    return(dispatch(setPhotoError()));
  }
};
