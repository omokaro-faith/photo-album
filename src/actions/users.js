import axios from 'axios'

export const setUsers = (users) => ({
 type: 'GET_USERS',
 users
});

export const getUsers = () => {
 return async dispatch => {
   try {
    const users = await axios.get('https://jsonplaceholder.typicode.com/users')
    return dispatch(setUsers(users.data));
   } catch (error) {
     console.log(error);
   }
 }
};
