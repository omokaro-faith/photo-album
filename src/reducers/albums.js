const initialState = {}

 export default (state= initialState, action) => {
 switch (action.type) {
   case 'GET_ALBUMS':
   const { albums } = action;
     return {...state, albums};
   case 'ALBUMS_ERROR':
   const { message } = action; 
      return { ...state, message}
   default:
     return state;
 }
}
