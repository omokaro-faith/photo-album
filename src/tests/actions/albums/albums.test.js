import configureMockStore from 'redux-mock-store'
import nock from 'nock';
import thunk from 'redux-thunk';
import { setAlbums, getAlbums } from '../../../actions/albums';
import albums from '../../fixtures/albums';

const createMockStore = configureMockStore([thunk]);

describe('setAlbums', () => {
  it('should set set albums', () => {
    const action = setAlbums(albums);

  expect(action).toEqual({
    type: 'GET_ALBUMS',
    albums,
  });
  });
});

describe('Async getAlbums', () => {
  let store;

  beforeEach(() => {
    store = createMockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('gets albums',async () => {
    nock('https://localhost:8000')
      .get('https://jsonplaceholder.typicode.com/albums?_start=0&_limit=')
      .reply(200);


    await store.dispatch(getAlbums())
      .then(() => {
        expect(store.getActions()[0].type).toEqual('GET_ALBUMS');
        expect(store.getActions()).toMatchSnapshot();
      })
      .catch(() => {
        expect(store.getActions()[0].type).toEqual('ALBUMS_ERROR');
        expect(store.getActions()[0].message).toEqual('Cannot fetch albums');
      });
  });
})