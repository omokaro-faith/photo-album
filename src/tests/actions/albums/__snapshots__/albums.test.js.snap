// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async getAlbums gets albums 1`] = `
Array [
  Object {
    "albums": Array [
      Object {
        "id": 1,
        "title": "quidem molestiae enim",
        "userId": 1,
      },
      Object {
        "id": 2,
        "title": "sunt qui excepturi placeat culpa",
        "userId": 1,
      },
      Object {
        "id": 3,
        "title": "omnis laborum odio",
        "userId": 1,
      },
      Object {
        "id": 4,
        "title": "non esse culpa molestiae omnis sed optio",
        "userId": 1,
      },
      Object {
        "id": 5,
        "title": "eaque aut omnis a",
        "userId": 1,
      },
      Object {
        "id": 6,
        "title": "natus impedit quibusdam illo est",
        "userId": 1,
      },
      Object {
        "id": 7,
        "title": "quibusdam autem aliquid et et quia",
        "userId": 1,
      },
      Object {
        "id": 8,
        "title": "qui fuga est a eum",
        "userId": 1,
      },
      Object {
        "id": 9,
        "title": "saepe unde necessitatibus rem",
        "userId": 1,
      },
      Object {
        "id": 10,
        "title": "distinctio laborum qui",
        "userId": 1,
      },
      Object {
        "id": 11,
        "title": "quam nostrum impedit mollitia quod et dolor",
        "userId": 2,
      },
      Object {
        "id": 12,
        "title": "consequatur autem doloribus natus consectetur",
        "userId": 2,
      },
      Object {
        "id": 13,
        "title": "ab rerum non rerum consequatur ut ea unde",
        "userId": 2,
      },
      Object {
        "id": 14,
        "title": "ducimus molestias eos animi atque nihil",
        "userId": 2,
      },
      Object {
        "id": 15,
        "title": "ut pariatur rerum ipsum natus repellendus praesentium",
        "userId": 2,
      },
      Object {
        "id": 16,
        "title": "voluptatem aut maxime inventore autem magnam atque repellat",
        "userId": 2,
      },
      Object {
        "id": 17,
        "title": "aut minima voluptatem ut velit",
        "userId": 2,
      },
      Object {
        "id": 18,
        "title": "nesciunt quia et doloremque",
        "userId": 2,
      },
      Object {
        "id": 19,
        "title": "velit pariatur quaerat similique libero omnis quia",
        "userId": 2,
      },
      Object {
        "id": 20,
        "title": "voluptas rerum iure ut enim",
        "userId": 2,
      },
    ],
    "type": "GET_ALBUMS",
  },
]
`;
